obstore/python/obstore/fsspec.py

```diff
@@ -296,3 +468,66 @@ def read(self, length: int = -1) -> Any:
             data = self.fs.cat_file(self.path, self.loc, self.loc + length)
             self.loc += length
         return data
+
+
+def register(protocol: str | list[str], asynchronous: bool = False) -> None:
+    """Dynamically register a subclass of AsyncFsspecStore for the given protocol(s).
+
+    This function creates a new subclass of AsyncFsspecStore with the specified
+    protocol and registers it with fsspec. If multiple protocols are provided,
+    the function registers each one individually.
+
+    Args:
+        protocol (str | list[str]): A single protocol (e.g., "s3", "gcs", "abfs") or
+            a list of protocols to register AsyncFsspecStore for.
+        asynchronous (bool, optional): If True, the registered store will support
+            asynchronous operations. Defaults to False.
+
+    Example:
+        >>> register("s3")
+        >>> register("s3", asynchronous=True)  # Registers an async-store for "s3"
+        >>> register(["gcs", "abfs"])  # Registers both "gcs" and "abfs"
+
+    Notes:
+        - Each protocol gets a dynamically generated subclass named
+          `AsyncFsspecStore_<protocol>`.
+        - This avoids modifying the original AsyncFsspecStore class.
+
+    """
+    # Ensure protocol is of type str or list
+    if not isinstance(protocol, str | list):
```

Comment:
Does this syntax work in 3.9?

Regardless, we should allow any iterable input, not just a `List`